src/Microsoft.OData.Client/IHttpClientHandlerProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="IHttpClientHandlerProvider.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Net.Http;
+
+namespace Microsoft.OData.Client
+{
+    /// <summary>
+    /// Provides the <see cref="HttpClientHandler"/> instance(s) used by OData Client
+    /// to execute requests.
+    /// </summary>
+    public interface IHttpClientHandlerProvider
+    {
+        /// <summary>
+        /// Returns an <see cref="HttpClientHandler"/> instance that OData Client should
+        /// used to execute a request. This method is called per request.
+        /// </summary>
+        /// <returns>The <see cref="HttpClientHandler"/> instance that will be used to execute the request.</returns>
+        /// <remarks>
+        /// It's possible that the returned <see cref="HttpClientHandler"/> may be mutated inside a request hook
+        /// e.g. via <see cref="HttpClientRequestMessage.Credentials"/>.
+        /// OData Client does not dispose the <see cref="HttpClientHandler"/> instance.
+        /// </remarks>
+        HttpClientHandler GetHttpClientHandler();

```

Comment:
Did we want to provide the context or args for this? It seems like it will be difficult for the implementer to know when to re-use a `HttpClientHandler` instance or when to provide a new one. (this is truly a discussion question, I think I don't know enough about `HttpClientHandler` and `HttpClient` to know the answer).

Reply:
I had passed the http request message at the beginning, and there was consideration to pass context with more info. But I removed it after Mike's suggestion to keep things simple for 7.x